recipes/cmake/3.x.x/conanfile.py

```diff
@@ -11,19 +10,40 @@ class CMakeConan(ConanFile):
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/Kitware/CMake"
     license = "BSD-3-Clause"
-    exports_sources = ["CMakeLists.txt"]
     generators = "cmake"
-    settings = "os", "arch", "build_type"
+    settings = "os", "arch", "compiler", "build_type"
+
+    options = {
+        "with_openssl": [True, False, "auto"],
+    }
+    default_options = {
+        "with_openssl": "auto",
+    }
 
     _source_subfolder = "source_subfolder"
     _cmake = None
 
     def _minor_version(self):
         return ".".join(str(self.version).split(".")[:2])
 
+    @property
+    def _with_openssl(self):
+        if self.options.with_openssl == "auto":
+            return self.settings.os != "Windows"
+        return self.options.with_openssl
+
     def configure(self):
         if self.settings.os == "Macos" and self.settings.arch == "x86":
             raise ConanInvalidConfiguration("CMake does not support x86 for macOS")
+        if self.settings.build_type != "Release":
+            raise ConanInvalidConfiguration("only Release build_type is supported")
+        if self.settings.compiler == "Visual Studio":
+            if self.settings.compiler.runtime != "MT":
```

Comment:
now I doubt this is the right way to do that: https://github.com/conan-io/conan-center-index/issues/1831

Reply:
@jgsogo @SSE4 
Would it be fine to just build all combinations?
- Build cmake in Release/Debug.
- Build cmake on MSVC with MT/MD/MDd.

I think this is the most general solution with maximum compatibility for the future.

Reply:
Yes, I think the best so far is to build all of them. Raising for a `--profile:build` will block all the others `build_reuqires` in the same graph...